M365/MDO/MDOThreatPolicyChecker.ps1

```diff
@@ -0,0 +1,840 @@
+            Write-Host "Exchange Online Powershell Module is required" -ForegroundColor Red
+            exit
+        } elseif ($exoConnection.count -eq 1) {
+            Write-Host " "
+            Write-Host "Connected to EXO"
+            Write-Host "Session details"
+            Write-Host "Tenant Id: $($exoConnection.TenantId)"
+            Write-Host "User: $($exoConnection.UserPrincipalName)"
+        } else {
+            Write-Host "You have more than one EXO sessions. Please use just one session" -ForegroundColor Red
+            exit
+        }
+
+        if (($graphConnection.TenantId) -ne ($exoConnection.TenantId) ) {
+            Write-Host "`nThe Tenant Id from Graph and EXO are different. Please use the same tenant" -ForegroundColor Red
+            exit
+        }
+    }
+
+    if ($PSCmdlet.ParameterSetName -eq "AppliedTenant") {
+        # Define the cmdlets to retrieve policies from and their corresponding policy types
+        $Cmdlets = @{
+            "Get-HostedContentFilterRule"                                               = "Anti-spam Policy"
+            "Get-HostedOutboundSpamFilterRule"                                          = "Outbound Spam Policy"
+            "Get-MalwareFilterRule"                                                     = "Malware Policy"
+
+            "Get-AntiPhishRule"                                                         = "Anti-phishing Policy"
+            "Get-SafeLinksRule"                                                         = "Safe Links Policy"
+            "Get-SafeAttachmentRule"                                                    = "Safe Attachment Policy"
+            "Get-ATPBuiltInProtectionRule"                                              = "Built-in protection preset security Policy"
+            { Get-EOPProtectionPolicyRule -Identity 'Strict Preset Security Policy' }   = "EOP"
+            { Get-EOPProtectionPolicyRule -Identity 'Standard Preset Security Policy' } = "EOP"
+            { Get-ATPProtectionPolicyRule -Identity 'Strict Preset Security Policy' }   = "MDO (Safe Links / Safe Attachments)"
+            { Get-ATPProtectionPolicyRule -Identity 'Standard Preset Security Policy' } = "MDO (Safe Links / Safe Attachments)"
+        }
+        $FoundIssues = $false
+
+        Write-Host " "
+
+        # Loop through each cmdlet
+        foreach ($Cmdlet in $Cmdlets.Keys) {
+            # Retrieve the policies
+            $Policies = & $Cmdlet
+
+            # Loop through each policy
+            foreach ($Policy in $Policies) {
+                # Initialize an empty list to store issues
+                $Issues = $null
+
+                # Check the logic of the policy and add issues to the list
+                if ($Policy.SentTo -and $Policy.ExceptIfSentTo) {
+                    $Issues += "`t`t-> User inclusions and exclusions. `n`t`t`tExcluding and including Users individually is redundant and confusing as only the included Users could possibly be included.`n"
```

Comment:
If you use a generic list, you can add them like this: `$Issues.Add("Your string goes here")`